From b7e1267f98a7ee7e571afa919a0489b4e2cc92f7 Mon Sep 17 00:00:00 2001
From: Randy Witt <randy.e.witt@linux.intel.com>
Date: Thu, 10 Sep 2015 11:12:29 -0700
Subject: [PATCH 3/3] zephyr-env.sh: Don't wait to test for error condition

Previously if a user had done "set -e" in their shell and ran this script
it was possible that "echo "${PATH}" | grep -q "${scripts_path}" could
cause the script to error out completely.

Instead of runnning the command then doing a test on $?, just run the
test on the command itself.

Signed-off-by: Randy Witt <randy.e.witt@linux.intel.com>
---
 zephyr-env.sh | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/zephyr-env.sh b/zephyr-env.sh
index 971be8d..a17a681 100644
--- a/zephyr-env.sh
+++ b/zephyr-env.sh
@@ -18,8 +18,9 @@ else
 fi
 
 scripts_path=${ZEPHYR_BASE}/scripts
-echo "${PATH}" | grep -q "${scripts_path}"
-[ $? != 0 ] && export PATH=${scripts_path}:${PATH}
+if ! echo "${PATH}" | grep -q "${scripts_path}"; then
+    export PATH=${scripts_path}:${PATH}
+fi
 unset scripts_path
 
 # enable custom environment settings
-- 
2.4.3

